### PR TITLE
rm duplicates nabi1, nabi2, fmptd2, mptex2, nfbii2

### PR DESCRIPTION
**Pull request by benjub**
Duplicates from #2128.
Updated credits as I could; not sure whether @sctfn's addition was shorter than the original, but at least now both Anthony Hart and him are mentioned.

### CODE REVIEW

**Comment by benjub** [comment]
Second commit: another duplicate from #2128.

**Comment by benjub** [comment]
Third commit: another duplicate from #2128. I hope it's good for @digama0 and @glacode. As for the label, it is "fvmptelrn", but as the comment says, it is "Function Value for function given in Maps-To notation is an ELement of codomain", and not "range". Is there an abbreviation for "codomain" that we use in labels ? Maybe "cd", since "dm" is for "domain" ?

File: set.mm [hunk 1]
```diff
@@ -12253,13 +12253,15 @@ This definition (in the form of ~ dfifp2 ) appears in Section II.24 of
     QUITUHUCTUIABIJUFUAUGUBAKBKLMNTUBUAOR $.
 
   $( Introduce a right anti-conjunct to both sides of a logical equivalence.
```
Comment (wlammen) [inline set.mm:12255]: I checked out the master of 01-01-2018 and looked into AH's mathbox.  It seems no shortening took place, but a renaming.  Here is what I found:
```
$( set.mm - Version of 2-Jan-2018 $)
```
...
```
$(
#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
                   Mathbox for Anthony Hart
#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
$)

$(
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
        Propositional Calculus
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
$)
```
...
```
  $( Constructor theorem for ` -/\ ` .  (Contributed by Anthony Hart,
     1-Sep-2011.) $)
  nabi1 $p |- ( ( ph <-> ps ) -> ( ( ph -/\ ch ) <-> ( ps -/\ ch ) ) ) $=
    ( wb wa wn wnan anbi1 notbid df-nan 3bitr4g ) ABDZACEZFBCEZFACGBCGLMNABCHIA
    CJBCJK $.

  $( Constructor theorem for ` -/\ ` .  (Contributed by Anthony Hart,
     1-Sep-2011.) $)
  nabi2 $p |- ( ( ph <-> ps ) -> ( ( ch -/\ ph ) <-> ( ch -/\ ps ) ) ) $=
    ( wb wa wn wnan anbi2 notbid df-nan 3bitr4g ) ABDZCAEZFCBEZFCAGCBGLMNABCHIC
    AJCBJK $.
```

Comment (benjub) [reply set.mm:12255]: Thanks for the enquiry @wlammen. What do you recommend ? Remove these two "proof shortened" tags ?

**Comment by benjub** [comment]
Fourth commit: deprecate duplicate nfbii2 from #2128.

File: set.mm [hunk 2]
```diff
@@ -12253,15 +12253,13 @@ This definition (in the form of ~ dfifp2 ) appears in Section II.24 of
...
-     2-Jan-2018.) $)
+     (Contributed by Anthony Hart, 1-Sep-2011.) $)
   nanbi1 $p |- ( ( ph <-> ps ) -> ( ( ph -/\ ch ) <-> ( ps -/\ ch ) ) ) $=
     ( wb wa wn wnan anbi1 notbid df-nan 3bitr4g ) ABDZACEZFBCEZFACGBCGLMNABCHIA
     CJBCJK $.
 
   $( Introduce a left anti-conjunct to both sides of a logical equivalence.
-     (Contributed by Anthony Hart, 1-Sep-2011.)  (Proof shortened by SF,
-     2-Jan-2018.) $)
+     (Contributed by Anthony Hart, 1-Sep-2011.) $)
```
Comment (wlammen) [inline set.mm:12262]: I am afraid, this one is a shortening.

**Review by wlammen — CHANGES_REQUESTED** [review]

**Review by wlammen — APPROVED** [review]
ok

File: set.mm [hunk 3]
```diff
@@ -59314,12 +59316,11 @@ in the range of the function (the implication "to the right" is always
...
-    $( If a class given in maps-to notation is a function with domain and
-       codomain, then its values are sets.  (Contributed by Glauco Siliprandi,
-       11-Dec-2019.) $)
-    mptex2 $p |- ( ( ph /\ t e. A ) -> B e. C ) $=
+    $d A x $.  $d C x $.
+    fvmptelrn.1 $e |- ( ph -> ( x e. A |-> B ) : A --> C ) $.
+    $( The value of a function at a point of its domain belongs to its
+       codomain.  (Contributed by Mario Carneiro, 29-Dec-2016.) $)
+    fvmptelrn $p |- ( ( ph /\ x e. A ) -> B e. C ) $=
       ( wcel cmpt wf wral eqid fmpt sylibr r19.21bi ) ADEGZBCACEBCDHZIOBCJFBCED
```
Comment (avekens) [inline set.mm:59324]: "rn" is often used as label fragment when "codomain" is meant, see ~ffvelrn and ~ffvelrnda. So we would have to chane this everywhere consistently. For now, we can keep this name.

**Comment by avekens** [comment]
> Third commit: another duplicate from #2128. I hope it's good for @digama0 and @glacode. As for the label, it is "fvmptelrn", but as the comment says, it is "Function Value for function given in Maps-To notation is an ELement of codomain", and not "range". Is there an abbreviation for "codomain" that we use in labels ? Maybe "cd", since "dm" is for "domain" ?

See my inline comment about this issue.

**Review by avekens — COMMENTED** [review]
PR approved.

**Comment by glacode** [comment]
> Third commit: another duplicate from #2128. I hope it's good for @digama0 and @glacode. As for the label, it is "fvmptelrn", 

I'm afraid there's a mistake in the "Contributed by" comment in fvmptelrn.

I noticed it was in my mathbox, thus I did a
git log -S "fvmptelrn"
and it looks like I committed it here
https://github.com/glacode/set.mm/commits/0fa1961cef2467dd69ae30b68366fa18b421d148

My best guess is that I modified a theorem created by @digama0 in 2016, to create fvmptelrn, but then I forgot to modify the "Contributed by" clause.

It looks like it should be

(Contributed by Glauco Siliprandi, 26-Jun-2021.)

I don't care for the attribution :-) but someone who searched for that theorem would not find it before June 2021, even though it states December 2016.

**Comment by glacode** [comment]
As a further (more important) note, we should have an alert in the PA that warns we are creating a theorem that's already in the theory.
I'm pretty sure I wrote mptex2 well before I wrote fvmptelrn , but I didn't find it due to its weird name and the "non-standard" use of
t e. A |->
in place of the usual
x e. A |->

As far as I can see, it shouldn't be hard to add such warning to the tool I'm working on (not a high priority feature, but nice to have in the future).
It should also be easy to add such control to mmj2, I guess.

**Comment by benjub** [comment]
> I'm afraid there's a mistake in the "Contributed by" comment in fvmptelrn.
> 
> I noticed it was in my mathbox, thus I did a git log -S "fvmptelrn" and it looks like I committed it here https://github.com/glacode/set.mm/commits/0fa1961cef2467dd69ae30b68366fa18b421d148
> 
> My best guess is that I modified a theorem created by @digama0 in 2016, to create fvmptelrn, but then I forgot to modify the "Contributed by" clause.
> 
> It looks like it should be
> 
> (Contributed by Glauco Siliprandi, 26-Jun-2021.)
> 
> I don't care for the attribution :-) but someone who searched for that theorem would not find it before June 2021, even though it states December 2016.

Where is this third date coming from ? mptex2 says "11-December-2019"

**Comment by benjub** [comment]
> As a further (more important) note, we should have an alert in the PA that warns we are creating a theorem that's already in the theory. I'm pretty sure I wrote mptex2 well before I wrote fvmptelrn , but I didn't find it due to its weird name and the "non-standard" use of t e. A |-> in place of the usual x e. A |->
> 
> As far as I can see, it shouldn't be hard to add such warning to the tool I'm working on (not a high priority feature, but nice to have in the future). It should also be easy to add such control to mmj2, I guess.

When adding a new theorem, I begin with `improve all /depth 2 /1` and once proved, I do at the end a `minimize_with *`. If this reduces the proof to one essential step, then I do not add the theorem.

**Comment by glacode** [comment]
> Where is this third date coming from ? mptex2 says "11-December-2019"

if you click here
https://github.com/glacode/set.mm/commit/0fa1961cef2467dd69ae30b68366fa18b421d148#diff-6774aa7a2ac0fc8c6ace3aaf14b7cf12a4bd3267395dc7c62d6733213d0d2174R637388

and "Load diff" it will bring you to the line where fvmptelrn is introduced.

In that commit, all the theorems (see the one just above and the one just below) have date 26-Jun-2021

**Comment by glacode** [comment]
> When adding a new theorem, I begin with `improve all /depth 2 /1` and once proved, I do at the end a `minimize_with *`. If this reduces the proof to one essential step, then I do not add the theorem.

This is a good suggestion, thank you. But I guess it would be nice to be warned even before doing the new (useless) proof.

(I'm not aware of the "improve all" feature, I'm gonna look into it)

**Comment by benjub** [comment]
Ok, fixed in #2811.
As for `improve`: definitely look at its help documentation. It's very useful: when a proof is "almost done", it is generally able to finish it, with its various options.  The best would be to have a tool able to combine the powers of mmj2 and MM-PA...

**Pull request merged**